kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -289,3 +407,224 @@ class LazyEBSDMasterPattern(EBSDMasterPattern, LazySignal2D):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+
+def _get_direction_cosines(detector: EBSDDetector) -> Vector3d:
+    """
+    Get the direction cosines between the detector and sample as done in EMsoft
+    and [Callahan2013]_.
+
+    Parameters
+    ----------
+    detector : EBSDDetector
+        EBSDDetector object with a certain detector geometry.
+
+    Returns
+    ----------
+    r_g.unit : Vector3d
+        Vector3d object containing the direction cosines for each detector
+        pixel.
+    """
+
+    pc = detector.pc_emsoft()
+    xpc = pc[..., 0]
+    ypc = pc[..., 1]
+    L = pc[..., 2]
+
+    # Detector coordinates in microns
+    det_x = (
+        -((-xpc - (1.0 - detector.ncols) * 0.5) - np.arange(0, detector.ncols))
+        * detector.px_size
+    )
+    det_y = (
+        (ypc - (1.0 - detector.nrows) * 0.5) - np.arange(0, detector.nrows)
+    ) * detector.px_size
+
+    # Auxilliary angle to rotate between reference frames
+    theta_c = np.radians(detector.tilt)
+    sigma = np.radians(detector.sample_tilt)
+
+    alpha = (np.pi / 2) - sigma + theta_c
+    ca = np.cos(alpha)
+    sa = np.sin(alpha)
+
+    # NYI
+    omega = np.radians(0)  # angle between normal of sample and detector
+    cw = np.cos(omega)
+    sw = np.sin(omega)
+
+    r_g_array = np.zeros((detector.nrows, detector.ncols, 3))
+
+    Ls = -sw * det_x + L * cw
+    Lc = cw * det_x + L * sw
+
+    i, j = np.meshgrid(
+        np.arange(detector.nrows - 1, -1, -1),
+        np.arange(detector.ncols),
+        indexing="ij",
+    )
+
+    r_g_array[..., 0] = det_y[i] * ca + sa * Ls[j]
+    r_g_array[..., 1] = Lc[j]
+    r_g_array[..., 2] = -sa * det_y[i] + ca * Ls[j]
+    r_g = Vector3d(r_g_array)
+
+    return r_g.unit
+
+
+def _get_lambert_interpolation_parameters(
+    rotated_direction_cosines: Vector3d,
+    scale: Union[int, float],
+    npx: int,
+    npy: int,
+) -> tuple:
+    """Get Lambert interpolation parameters as described in EMsoft.
+
+    Parameters
+    ----------
+    rotated_direction_cosines : Vector3d
+        Rotated direction cosines vector.
+    scale : int
+        Factor to scale up from Rosca-Lambert projection to the master pattern.
+    npx : int
+        Number of pixels on the master pattern in the x direction.
+    npy : int
+        Number of pixels on the master pattern in the y direction.
+
+    Returns
+    ----------
+    nii : numpy.ndarray
+        Row coordinate of a point.
+    nij : numpy.ndarray
+        Column coordinate of a point.
+    niip : numpy.ndarray
+        Row coordinate of neighboring point.
+    nijp : numpy.ndarray
+        Column coordinate of a neighboring point.
+    di : numpy.ndarray
+        Interpolation weight factor.
```

Comment:
The return descriptions of the other parameters are nice and informative, but it would be nice to specify which coordinates these are weight factors for.